### PR TITLE
Adding some basic pytest

### PR DESCRIPTION
**Pull request by Lash-L**
Super basic test so far, but it is a basis to build on.

Also added ci pipeline so that when a PR gets made or code is pushed into dev it will run all of the test, to make sure there's never an error introduced like I did with the circular dependency.

I can also add a ci task to automatically release a new version on pypi on dev push if that is something you are interested in.

You will have to enable Actions on github if this is something you want

This also caught that Python 3.9 is not supported(which doesn't matter as HA only allows for 3.10 and above

### CODE REVIEW

File: tests/test_containers.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    assert device.pv == "1.0"
+    assert device.room_id == 2362003
+    assert device.tuya_uuid is None
+    assert not device.tuya_migrated
+    assert device.extra == '{"RRPhotoPrivacyVersion": "1"}'
+    assert device.sn == "abc123"
+    assert device.feature_set == "2234201184108543"
+    assert device.new_feature_set == "0000000000002041"
+    # status = device.device_status
+    # assert status.name ==
```
Comment (Lash-L) [inline tests/test_containers.py:76]: device_status is not actually correct. The parameters for it are completely different than what the api returns. I don't think it used anywhere, but I noticed it from the test so I figured I would share.

**Comment by Lash-L** [comment]
@humbertogontijo Let me know when you end up pushing the new version to pypi. Alternatively, let me know if you want my help automating it with ci/cd

**Comment by Lash-L** [comment]
@humbertogontijo ill have some time to work on this today, do you think you'll have a chance to publish to pypi?

**Comment by Lash-L** [comment]
@humbertogontijo Thanks for merging and pushing to pypi! I'm going to create the HA PRs shortly.

Let me know if you have any questions. I marked you as a codeowner.

feel free to email me conway220@gmail.com if you have any questions.

**Comment by humbertogontijo** [comment]
Hey! Thank you for your effort. 
I'm so sorry for the delay reviewing your changes, I'm currently short in time for hobbies repositories, but I'll do my best to keep up to date with the roborock integration.

Thanks again for all your help

**Comment by humbertogontijo** [comment]
Ah, send me the link for your PR to HA core when you have it please

**Comment by Lash-L** [comment]
@humbertogontijo no worries, I completely get it.
Here's the PR!

https://github.com/home-assistant/core/pull/89456

**Pull request merged**